### PR TITLE
net: Rework neighbour

### PR DESCRIPTION
**Pull request by anton-bondarev**

### CODE REVIEW

File: src/net/neighbour.c [hunk 1]
```diff
@@ -362,68 +349,39 @@ int neighbour_foreach(neighbour_foreach_ft func, void *args) {
 	return 0;
 }
 
-int neighbour_send_after_resolve(unsigned short ptype,
+int neighbour_resolve(unsigned short ptype,
 		const void *paddr, unsigned char plen,
-		struct net_device *dev, struct sk_buff *skb) {
-	int allocated, resolved;
+		struct net_device *dev, struct sk_buff *skb,
+		unsigned char hlen_max, void *out_haddr) {
```
Comment (alexkalmuk) [inline src/net/neighbour.c:355]: As far as I can see `hlen_max` is unused in this function.

Comment (alexkalmuk) [reply src/net/neighbour.c:355]: Btw, it looks like we can remove `hlen` as a parameter, also in neighbour_add for example.

Comment (anton-bondarev) [reply src/net/neighbour.c:355]: Thanks, it's out_haddr length.
I think It must be checked that the buffer size is enough for dev->addr (dev->addr_len)

Comment (alexkalmuk) [reply src/net/neighbour.c:355]: Ok, please check also neighbour_add. There we also fill haddr but do not rely on hlen (only checking for `(hlen > sizeof(nbr->haddr`)

Comment (anton-bondarev) [reply src/net/neighbour.c:354]: thanks, done

**Pull request merged**